PlayList.java

```diff
@@ -35,32 +35,59 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
+        int size1 = this.getSize();
+        if(size1 == this.getMaxSize()){
+            return false;
+        }
+        else{
+            tracks[size1] = track;
+            size = size1 + 1;
+        }
```

Comment:
No need `else` when the `if` scope return and escape method - the `else` part won't be never reached.
```suggestion
            tracks[this.size] = track;
            this.size++;
```